### PR TITLE
Add `Hash#force_compact`

### PR DESCRIPTION
**Pull request by yhirano55**
`Hash#force_compact` returns a hash with non blank values.

example:

``` ruby
hash = { a: true, b: false, c: nil, d: [], e: {}, f: ''}
hash.force_compact # => { a: true}
hash # => { a: true, b: false, c: nil, d: [], e: {}, f: ''}
{ b: false, c: nil, d: [], e: {}, f: '' }.force_compact # => {}
```

### CODE REVIEW

**Comment by maclover7** [comment]
I'm not sure if we take PRs to add to Active Support's `core_ext` extensions... cc @rafaelfranca

**Comment by kaspth** [comment]
Thanks for the pull request! We only add methods to Active Support if they prove to be generally useful to most apps, and your proposal, while well written, I don't think fits that bill. :heart:

**Comment by EdDeAlmeidaJr** [comment]
Sorry @kaspth, but may I respectfully ask what are the criteria to a method be "generally useful to most apps"? It is just a curiosity. Good to know this, so people won't spend too much effort in non acceptable things, don't you think.

**Comment by rafaelfranca** [comment]
TL;DR at this point almost anything is useful to most applications.

The criteria we use is: Do we see using this in our application?

If the answer is yes we merge, but if the answer is no we don't. It is really hard to find any new extension that match this criteria so I prefer to say that core extension are frozen now unless added by people at Rails core.

**Comment by kaspth** [comment]
@EdDeAlmeidaJr good question! @rafaelfranca answered it before I could form one myself 😁

**Comment by EdDeAlmeidaJr** [comment]
Thanks @rafaelfranca and @kaspth! Much better this way. Transparency, as those politically correct people like to say, so people in general won't spend precious time. There is always so much to be done, as we all know.
